ReactiveUI.Fody/ReactiveUIPropertyWeaver.cs

```diff
@@ -54,6 +58,59 @@ public void Execute()
             {
                 foreach (var property in targetType.Properties.Where(x => x.IsDefined(reactiveAttribute)).ToArray())
                 {
+                    TypeReference genericTargetType = targetType;
+                    if (targetType.HasGenericParameters)
+                    {
+                        var genericDeclaration = new GenericInstanceType(targetType);
+                        foreach (var parameter in targetType.GenericParameters)
+                        {
+                            genericDeclaration.GenericArguments.Add(parameter);
+                        }
+                        genericTargetType = genericDeclaration;
+                    }
+
+                    if (property.SetMethod == null && property.GetMethod.IsExpressionBodiedProperty())
+                    {
+                        // Add this.RaisePropertyChange(PropertyName) to the setter of any properties referenced 
+                        // by this expression bodied property where they have an available setter
+                        var getInstructions = property.GetMethod.Body.Instructions.Where(i =>
+                        {
+                            var operand = i.Operand as MethodDefinition;
+                            return i.OpCode == OpCodes.Call && (operand?.IsGetter ?? false);
+                        });
+
+                        var setMethodsForGetInstructions =
+                            targetType.Properties.Where(
+                                    prop =>
+                                        prop.SetMethod != null &&
+                                        prop.SetMethod.HasBody &&
+                                        getInstructions.Any(
+                                            i => ((MethodDefinition) i.Operand).FullName == prop.GetMethod.FullName))
+                                .Select(prop => prop.SetMethod);
+
+                        if (!setMethodsForGetInstructions.Any())
+                        {
+                            LogError(
+                                $"Property {property.DeclaringType.FullName}.{property.Name} only references other expression bodied properties, as these have no setter they're not supported by the [Reactive] attribute");
+                        }
+
+                        var raisePropertyChangedMethodReference = raisePropertyChangedMethod.MakeGenericMethod(genericTargetType);
+
+                        foreach (var method in setMethodsForGetInstructions)
+                        {
+                            method.Body.Emit(il =>
+                            {
+                                var last = method.Body.Instructions.Last(i => i.OpCode == OpCodes.Ret && (i.Next == null || method.Body.Instructions.Last() == i));
```

Comment:
I believe all methods (even void methods) must have an `OpCodes.Ret`, so probably the secondary check is not necessary?  However, I wonder what would happen if you had a property set method implemented like:

```
public string Foo
{
    get => foo;
    set {
        if (bar)
        {
            this.foo = value;
            return;
        }
        this.foo = "baz";
    }
}
```

I think your PR will fail to propagate the `RaisePropertyChange` when `bar` is `true` since this method contains two returns and you're only adding your logic to the last one.  A little annoying, but it'd probably make sense to get all the return instructions and instrument the below three lines in front of each.